Services/DeviceModels.cs

```diff
@@ -143,6 +149,35 @@ public class DeviceModels : IDeviceModels
             await this.customDeviceModels.DeleteAsync(id);
         }
 
+        /// <summary>
+        /// Get property names from all device models.
+        /// </summary>
+        public async Task<List<DeviceProperty>> GetPropertyNamesAsync()
+        {
+            /*
+             * This will only return the initial device properties that are listed in the Device Model,
+             * this will not return all of the properties that are actually associated with the device
+             * after start-up.
+             * The properties added after startup will come from the new API located in IOT Hub Manager.
+             * The UI will merge both these results.
+             * */
+            var list = await this.GetListAsync();
```

Comment:
Can we query IoTHub here to get a list of all properties instead of returning partial list ?

Reply:
And also properties from twin ?

Reply:
As discussed this is specific for device models and not for all devices